src/plugins/intel_gna/src/ops/util/util.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "util.hpp"
+
+using namespace GNAPluginNS::GNALimitations;
+using namespace ov::intel_gna;
+
+namespace {
+std::ostream& operator<<(std::ostream& os, const std::set<ov::element::Type>& t) {
+    for (auto it = t.begin(); it != t.end(); ++it) {
+        if (it != t.begin()) {
+            os << ", " << *it;
+        } else {
+            os << *it;
+        }
+    }
+    return os;
+}
+
+bool is_precision_supported(const std::shared_ptr<ov::Node>& node,
+                            const std::set<ov::element::Type>& supported_types,
+                            bool is_exception_allowed) {
+    if (!node) {
+        if (is_exception_allowed) {
+            THROW_GNA_EXCEPTION << "Node is empty!\n";
+        }
+        return false;
+    }
+    if (supported_types.count(node->get_element_type()) == 0) {
+        if (is_exception_allowed) {
+            THROW_GNA_EXCEPTION << "The plugin does not support input precision with " +
+                std::string(node->get_type_name()) + " format. Supported input precisions " << supported_types << "\n";
+        }
+        return false;
+    }
+    return true;
+}
+
+bool is_conv_supported(const std::shared_ptr<ngraph::op::ConvolutionIE>& conv_ie,
```

Comment:
I just wonder if this method could be put in some common place where other convolution related methods are placed.
e.g.
https://github.com/marcinkusm/openvino/blob/master/src/plugins/intel_gna/layers/gna_convolution_layer.hpp

Reply:
My bad, I see that layers subdirectory is related to the legacy layers.